src/view-backend-exportable-fdo-egl.cpp

```diff
@@ -285,16 +289,20 @@ class ClientBundleEGL final : public ClientBundle {
     {
         assert(image->eglImage);
         WS::instanceImpl<WS::ImplEGL>().destroyImage(image->eglImage);
-
         delete image;
     }
 
     static void bufferDestroyListenerCallback(struct wl_listener* listener, void*)
     {
         struct wpe_fdo_egl_exported_image* image;
         image = wl_container_of(listener, image, bufferDestroyListener);
-
-        image->bufferResource = nullptr;
+        // The image object is inmediately destroyed here only if it's not currently
+        // exported, otherwise it's destroyed when the client releases/returns it.
+        if (image->exported) {
+            image->exported = false;
```

Comment:
Good. This forces entering in the  `deleteImage(image)` after in the `releaseImage(struct wpe_fdo_egl_exported_image* image)`.